scene/gui/code_edit.h

```diff
@@ -377,12 +382,19 @@ class CodeEdit : public TextEdit {
 	void fold_line(int p_line);
 	void unfold_line(int p_line);
 	void fold_all_lines();
+	void create_code_region();
 	void unfold_all_lines();
 	void toggle_foldable_line(int p_line);
 
 	bool is_line_folded(int p_line) const;
 	TypedArray<int> get_folded_lines() const;
 
+	/* Code region */
```

Comment:
Using `open / close` sounds better to me then `opening / closing`.

Reply:
> Reviewed `/scene` changes.
> 
> Not sure on making the editor side quite so advanced yet as I would like to refactor that first, might be worth splitting those changes out into a separate PRs.

Thanks for you review, I'm working on it with your comments. For the editor part, maybe we could remove from this PR the replacement of the item list by the tree (if that's the part of editor you want to refactor first) and keep only "Create code region" menu item and shortcut.

For delimiters, it will now choose the first one-line comment delimiter (not pushed to github yet). I think it's good enough. All languages seems to have a one line delimiter. Supporting an endkey comment delimiter would make code more complex for a use case that seems more than rare. If someone with such a language try to create a region, it will just warn that there we can't without a oneline comment defined.